### PR TITLE
Feature/cc 0051 canvas & court responsive design

### PR DESCRIPTION
**Pull request by Beryl-Zhou**
94ef80d297f04681939b437149bc168
86a6e0d6709248024196aee29d4b89c
b7b6276d9199759c3cec18ad52cb7ed

### CODE REVIEW

**Comment by stkwn** [comment]
When the browser height is smaller than the width, there seems to be a overlay when using chrome.

File: src/components/BasketballCourt/Arrow.tsx [hunk 1]
```diff
@@ -3,16 +3,24 @@ import { useStoreSelector } from "@/store/hooks";
...
+  arrowXEndLength,
+  initPointX,
+  initPointY,
+}) => {
+  const { courtAreaXLength, courtAreaYLength, borderLength } = useStoreSelector(
+    (state) => state.courtSize
+  );
   let border;
-  borderLength <= 100 ? (border = 100 * courtRatio) : (border = borderLength * courtRatio);
+  borderLength <= 1000 ? (border = 1000 * courtRatio) : (border = borderLength * courtRatio);
```
[the remark this answers is not in the document]
Comment (Beryl-Zhou) [reply src/components/BasketballCourt/Arrow.tsx:21]: Got it~~~ other conversations with a similar scenario will be updated together. Cheers~~

File: src/components/BasketballCourt/Arrow.tsx [hunk 2]
```diff
@@ -3,16 +3,24 @@ import { useStoreSelector } from "@/store/hooks";
 
 interface ArrowProps {
   courtRatio: number;
+  initPointX: number;
+  initPointY: number;
```
[the remark this answers is not in the document]
Comment (Beryl-Zhou) [reply src/components/BasketballCourt/Arrow.tsx:7]: Got it~~

**Comment by kritomomo** [comment]
@chui-mel Just my suggestion 🙈, because it will be easier to implement this feature, no need to think about different sizes anymore.. Also this idea is same to the zoom in/out, except that one is manual control and this one is automatic adaptation. It's up to Beryl, she can try this one and compare this two solutions

File: src/components/ProFullCourt/index.tsx [hunk 3]
```diff
@@ -10,20 +11,50 @@ import Border from "../BasketballCourt/Border";
...
+        height: window.innerHeight,
+      });
+    };
+
+    window.addEventListener("resize", checkSize);
+    return () => window.removeEventListener("resize", checkSize);
+  }, []);
+
+  const stageMargin = 2500;
+  const convaWidth = courtAreaXLength * 2 + stageMargin * 2; // actual court size plus reserved margin size (prepare for 2m border)
```
Comment (Fuchih) [inline src/components/ProFullCourt/index.tsx:33]: why "conva" not "canvas", just curious

**Comment by tatoMa** [comment]
> image
> 
> My idea is that we can set an initial width and height for this canvas, and then use the scale of the stage to zoom in and out. And how to get this specific scale value, taking the width as an example, we can use **(window.innerWidth - the width of the left sidebar (98px)) / 850,** this is our scaleX, and then the width of this stage and the minWidth of the surrounding flex are equal to *_850 _scaleX__.
> 
> About how to get the latest value when we adjust the browser, we can use useEffect, write a function, use setState to get the current window.innerWidth in real time, and then combine https://developer.mozilla.org/en-US/docs/Web/API/Window /resize_event
> 
> In addition, you can also consider whether the size of the canvas should start to shrink after the page is zoomed to a certain extent, and to what extent it will stop shrinking. image
> 
> image

My first idea for this feature comes up just like yours. Easier to implement and fits all resolutions and different ratios. Nice one.⭐️

**Review by tatoMa — COMMENTED** [review]
I think it is better to abstract some of the calculations into separate utilities for: 
1. easier to read as they are a little too much and long in a UI component; 
2. might need to use these calculations for testing and other similar scenarios later( don't forget we have more than 1 size of court)

**Pull request closed without merging**